container/container_unix.go

```diff
@@ -280,18 +281,50 @@ func (container *Container) UnmountSecrets() error {
 
 // UpdateContainer updates configuration of a container.
 func (container *Container) UpdateContainer(hostConfig *containertypes.HostConfig) error {
+	sysInfo := sysinfo.New(true)
+
 	container.Lock()
 	defer container.Unlock()
 
 	// update resources of container
 	resources := hostConfig.Resources
 	cResources := &container.HostConfig.Resources
+
+	// validate NanoCPUs, CPUPeriod, and CPUQuota
+	if resources.NanoCPUs > 0 && resources.CPUPeriod > 0 {
```

Comment:
Thanks @thaJeztah for the review. The PR has been updated and duplicated parts has been removed.

The check for 
```
+	if resources.NanoCPUs > 0 && cResources.CPUPeriod > 0 {
 +		return fmt.Errorf("Conflicting options: Nano CPUs cannot be updated as CPU Period has already been set")
 +	}
...
```
are left alone. The reason is that, when updating CPUs, we could either update through `NanoCPU`, or through `CPUPeriod/CPUQuota`. However, once we update through one method (e.g., through `NanoCPU`), we cannot update through another method (e.g., through `CPUPeriod/CPUQuota`). For that reason, the check will need to be done by checking which method was used.